src/ResourceManagement/HDInsight/HDInsight/Customizations/ClusterOperations.Create.Customizations.cs

```diff
@@ -570,12 +570,29 @@ private static IEnumerable<Role> GetRoleCollection(ClusterCreateParameters clust
 
             roles.Add(zookeepernode);
 
+            if (clusterCreateParameters.ClusterType.Equals("RServer", StringComparison.OrdinalIgnoreCase))
+            {
+                var EdgeNodeSize = GetEdgeNodeSize(clusterCreateParameters);
+                var EdgeNode = new Role
+                {
+                    Name = "edgenode",
```

Comment:
I think this is fine for now. I'll do the pass through the SDK to clean it up separately, but for this PR, he is following the current convention.